freva-rest/src/databrowser_api/core.py

```diff
@@ -753,6 +764,28 @@ async def init_stream(self) -> Tuple[int, SearchResult]:
             primary_facets=[],
         )
 
+    def _join_facet_queries(
+        self, key: str, facets: List[str]
+    ) -> Tuple[str, str]:
+        """Create lucene search contain and NOT contain search queries"""
+
+        negative, positive = [], []
+        for search_value in facets:
+            if key not in self.uniq_keys:
+                search_value = search_value.lower()
+            if search_value.lower().startswith("not "):
+                search_value = "-" + search_value.lstrip("NOT ").lstrip("not ")
```

Comment:
should not be
```python
                negative.append(search_value.lstrip("NOT ").lstrip("not "))
```
instead ?

Reply:
Yes, I know the logic is kind of awkward. The idea is to replace leading *not*'s by leading -. I guess I could aswell do it like this:

```python
if search_value.lower().startswith("not "):
    negative.append(search_value.lstrip("NOT ").lstrip("not "))
elif search_value[0] in ("!", "-"):
     negative.append(search_value.lstrip("!").lstrip("-"))
else:
     positive.append(search_value)

```